### PR TITLE
Improve calendardataservice teardown, add QList guards for robustness

### PR DESCRIPTION
**Pull request by pgerdt**

### CODE REVIEW

File: src/calendarmanager.cpp [hunk 1]
```diff
@@ -224,12 +224,13 @@ QString NemoCalendarManager::getNotebookColor(const QString &notebookUid) const
 
 void NemoCalendarManager::cancelAgendaRefresh(NemoCalendarAgendaModel *model)
 {
-    mAgendaRefreshList.removeOne(model);
+    if (!mAgendaRefreshList.isEmpty())
+        mAgendaRefreshList.removeOne(model);
```
Comment (pvuorela) [inline src/calendarmanager.cpp]: Should this second commit have any effect? Removing not found item should already just return false and empty list will not contain any value.

**Comment by pvuorela** [comment]
LGTM on calendar data service commit.

File: src/calendarmanager.cpp (hunk 1 above)
Comment (pgerdt) [inline src/calendarmanager.cpp]: Well, the QList documentation says that isEmpty should be checked before calling member functions, see last paragraph of this section: http://doc.qt.io/qt-5/qlist.html#details

Comment (rburchell) [inline src/calendarmanager.cpp]: That particular reference is talking about things like .at(), and so on. removeOne is essentially "a for loop plus a return once it's removed the item", so it's safe without.

**Comment by pgerdt** [comment]
Updated PR: removed commit adding unnecessary QList checks.

**Comment by pvuorela** [comment]
Still looking good.

**Pull request merged**